cogs/detect_code.py

```diff
@@ -71,15 +74,25 @@ def likely_contains_code(text: str) -> bool:
             return False
 
         non_blank_lines = [line for line in lines if line.strip()]
+
+        patterns = [
+            r"\b\w+\s*\(\s*.*?\s*\)\s*",
+            r"(#.*$)",
+            r"'''",
+            r'"""',
+            r";(?=\s*(//|#).+|$)",
+            r"^[\w]+\.[\w]+(\.[\w]+)*$",
+            r"\b\w+(?:_\w+)*\b(?:->\b\w+(?:_\w+)*\b)+",
+            r"^\/\*",
+            r".*\*/$",
+            r"[A-Za-z]+(?:[a-z][A-Z]|[A-Z][a-z])[A-Za-z]*",
+            r"^\w+(?:_\w+)+$",
+            r"\b[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)+\b",
+        ]
```

Comment:
Some suggestions to simplify these patterns, in the context of how they're used:
1.`\b\w+\s*\(\s*.*?\s*\)\s*` is equivalent to `\w+\s*\(.*\)` 
2. `(#.*$)` is equivalent to `#` - any line containing a hashtag
3. `;(?=\s*(//|#).+|$)` is equivalent to `;\s*(//|#)`
4. `^[\w]+\.[\w]+(\.[\w]+)*$` is equivalent to `^\w+(\.\w+)+$`
5. `\b\w+(?:_\w+)*\b(?:->\b\w+(?:_\w+)*\b)+` is equivalent to `\w+(->\w+)+`
6. `.*\*/$` is equivalent to `\*/$`
7. `[A-Za-z]+(?:[a-z][A-Z]|[A-Z][a-z])[A-Za-z]*` is equivalent to `[A-Za-z]+([a-z][A-Z]|[A-Z][a-z])` and I would say that `[a-z][A-Z]` would be good enough on its own

Reply:
3 is an incorrect fix as it does not account for there being no comment after the semicolon. Everything else seems reasonable.

Reply:
I see 👍